test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -12480,6 +12481,38 @@ def forward(self, x):
 
         self.run_test(LerpModel(), torch.rand(5, 4, 3))
 
+    @common_utils.parametrize("use_complex_dtypes", [True, False])
+    @skipIfUnsupportedMinOpsetVersion(9)
+    def test_print_tensor_within_torch_nn_module(self, use_complex_dtypes: bool):
+        class PrintTensorOnMyModel(torch.nn.Module):
+            def forward(self, x):
+                x_firsts = x[:, 0]
+                print(f"x_firsts: {x_firsts}")
+                return x_firsts
+
+        m = PrintTensorOnMyModel().eval()
+        if use_complex_dtypes:
+            dtype = torch.cfloat
+        else:
+            dtype = torch.float
+        x = torch.randn(10, 5, dtype=dtype)
```

Comment:
For future reference, I would just parameterize the test with the two different dtypes as inputs and name the sub-tests. This way we avoid an if-else in the test body and the test can be made cleaner.